### PR TITLE
Build system: Wrap clang-format to show diff

### PR DESCRIPTION
**Pull request by julianoes**
This adds a bash script around clang format which allows us to show a diff if the formatting has changed.

### CODE REVIEW

**Review by LorenzMeier — APPROVED** [review]

**Comment by LorenzMeier** [comment]
Awesome!

File: tools/run_clang_format.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    diff_cmd=diff
+fi
+
+# Keep track of errors for the exit value
+error_found=false
+
+# Go through files and run clang-format in place and show diff against .orig file.
+for file in $@; do
+
+    cp $file $file.orig
```
Comment (bkueng) [inline tools/run_clang_format.sh:26]: @julianoes can we move this to something else, preferably in `/tmp`? `.orig` files might already exist and are used for different purpuses by another tool.

File: tools/run_clang_format.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    if ! cmp $file $file.orig >/dev/null 2>&1; then
+        echo "Changed $file:"
+        $diff_cmd $file.orig $file
+        error_found=true
+    fi
+
+    rm $file.orig
+done
+
+if [ "$error_found" = true ]; then
```
Comment (bkueng) [inline tools/run_clang_format.sh:38]: Would be simpler to define a `return_code` and then just do `exit $return_code`.

File: tools/run_clang_format.sh (hunk 1 above)
Comment (julianoes) [reply tools/run_clang_format.sh:26]: Good point.

File: tools/run_clang_format.sh (hunk 2 above)
Comment (julianoes) [reply tools/run_clang_format.sh:38]: Right.

**Pull request merged**